lib/cylc/graph_parser.py

```diff
@@ -161,13 +171,27 @@ def parse_graph(self, graph_string):
         """
         # Strip comments, whitespace, and blank lines.
         non_blank_lines = []
+        bad_lines = []
         for line in graph_string.split('\n'):
-            line = self.__class__.REC_COMMENT.sub('', line)
-            # Apparently this is the fastest way to strip all whitespace!:
-            line = "".join(line.split())
-            if not line:
+            modifiedLine = self.__class__.REC_COMMENT.sub('', line)
```

Comment:
Please use `modified_line`.

https://www.python.org/dev/peps/pep-0008/#function-and-variable-names